examples/conversation_with_RAG_agents/rag_example.py

```diff
@@ -1,63 +1,145 @@
 
 import agentscope
 from agentscope.agents import UserAgent
-from agentscope.message import Msg
-from agentscope.agents import DialogAgent
+from agentscope.rag import KnowledgeBank
+
+
+AGENT_CHOICE_PROMPT = """
+There are following available agents. You need to choose the most appropriate
+agent(s) to answer the user's question.
+
+agent descriptions:{}
+
+First, rephrase the user's question, which must contain the key information.
+The you need to think step by step. If you believe some of the agents are
+good candidates to answer the question (e.g., AGENT_1 and AGENT_2), then
+you need to follow the following format to generate your output:
+
+'
+Because $YOUR_REASONING.
+I believe @AGENT_1 and @AGENT_2 are the most appropriate agents to answer
+your question.
+'
+"""
+
+
+def prepare_docstring_html() -> None:
+    """prepare docstring in html for API assistant"""
+    if not os.path.exists("../../docs/docstring_html/"):
+        os.system(
+            "sphinx-apidoc -f -o ../../docs/sphinx_doc/en/source "
+            "../../src/agentscope -t template",
+        )
+        os.system(
+            "sphinx-build -b html  ../../docs/sphinx_doc/en/source "
+            "../../docs/docstring_html/ -W --keep-going",
+        )
 
 
 def main() -> None:
     """A RAG multi-agent demo"""
-    agentscope.init(
-        model_configs=[
-            {
-                "model_type": "dashscope_chat",
-                "config_name": "qwen_config",
-                "model_name": "qwen-max",
-                "api_key": f"{os.environ.get('DASHSCOPE_API_KEY')}",
-            },
-            {
-                "model_type": "dashscope_text_embedding",
-                "config_name": "qwen_emb_config",
-                "model_name": "text-embedding-v2",
-                "api_key": f"{os.environ.get('DASHSCOPE_API_KEY')}",
-            },
-        ],
-    )
+    # prepare html for api agent
+    prepare_docstring_html()
+
+    # prepare models
+    with open("configs/model_config.json", "r", encoding="utf-8") as f:
+        model_configs = json.load(f)
+    # for internal API
+    for config in model_configs:
+        if config.get("model_type", "") == "post_api_chat":
+            config["headers"]["Authorization"] = (
+                "Bearer " + f"{os.environ.get('HTTP_LLM_API_KEY')}"
+            )
+        else:
+            # for dashscope
+            config["api_key"] = f"{os.environ.get('DASHSCOPE_API_KEY')}"
 
-    with open("./agent_config.json", "r", encoding="utf-8") as f:
+    # load config of the agents
+    with open("configs/agent_config.json", "r", encoding="utf-8") as f:
         agent_configs = json.load(f)
-    tutorial_agent = LlamaIndexAgent(**agent_configs[0]["args"])
-    code_explain_agent = LlamaIndexAgent(**agent_configs[1]["args"])
-    summarize_agent = DialogAgent(**agent_configs[2]["args"])
+
+    agent_list = agentscope.init(
+        model_configs=model_configs,
+        agent_configs=agent_configs,
+    )
```

Comment:
Shall we assume that the api keys have been set in the config? Thus, the code can be cleaner as 
```
agent_list = agentscope.init(
        model_configs="./configs/model_confi.json",
        agent_configs="./configs/agent_config.json",
    )
```

Reply:
update as suggested